packages/runtime-common/virtual-network.ts

```diff
@@ -67,6 +115,73 @@ export class VirtualNetwork {
         ? urlOrRequest
         : new Request(urlOrRequest, init);
 
+    let internalRequest = await this.mapRequest(request, 'virtual-to-real');
+    let response = await this.runFetch(internalRequest, init);
+    if (internalRequest !== request) {
+      Object.defineProperty(response, 'url', {
+        value:
+          this.resolveURLMapping(response.url, 'real-to-virtual') ??
+          response.url,
+      });
+    }
+    return response;
+  };
+
+  async handle(
```

Comment:
Add a comment explaining what this method should be used for.

Reply:
Added a comment.

Reply:
I don't see the comment.